### PR TITLE
add Slack notification if cronjob daily build fails

### PR DESCRIPTION
**Pull request by stevesloka**
Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: .github/workflows/cronbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      - name: generate
+        run: |
+          make generate generate-contour-crds example manifests 
+          ./hack/actions/check-uncommitted-codegen.sh
+      - uses: act10ns/slack@v1
+        with:
+          status: ${{ job.status }}
+          steps: ${{ toJson(steps) }}
+          channel: '#contour-ci-notifications'
+        if: failure()
```
Comment (sunjayBhatia) [inline .github/workflows/cronbuild.yaml:183]: might be able to keep it in the original file if we can check the branch as well? (e.g. `if failure && ${{ github.head_ref }} == main` or something)

might make it so the scheduled runs and the `main` runs both alert

Comment (stevesloka) [reply .github/workflows/cronbuild.yaml:183]: yup that sounds good. Was looking for a way to not alert on each PR.

Comment (stevesloka) [reply .github/workflows/cronbuild.yaml]: ok I think my final commit implements this properly. I just. did a run and did not get notified, so I think the next build would notify.

**Review by sunjayBhatia — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
lets try it out! worth adding to the `main` and release branch builds in the Contour repo as well I think

**Pull request merged**